engine/univGen.ml

```diff
@@ -17,15 +17,17 @@ type univ_length_mismatch = {
   actual : int ;
   expect : int ;
 }
-(* Due to an OCaml bug ocaml/ocaml#10027 inlining this record will cause
-compliation with -rectypes to crash. *)
-exception UniverseLengthMismatch of univ_length_mismatch
 
-let () = CErrors.register_handler (function
-  | UniverseLengthMismatch { actual; expect } ->
-      Some Pp.(str "Universe instance length is " ++ int actual
-        ++ str " but should be " ++ int expect ++ str ".")
-  | _ -> None)
+type _ CErrors.tag += UniverseLengthMismatch : univ_length_mismatch CErrors.tag
```

Comment:
Since we arent using rectypes any longer we should inline.

Reply:
We can't inline, unless you mean use a anonymous tuple instead of record.

Reply:
Yes that is what I mean

Reply:
Record is nicer imo